homeassistant/components/freebox/camera.py

```diff
@@ -0,0 +1,160 @@
+
+        super().__init__(hass, router, node)
+        self._flip: Any
+        device_info = {
+            CONF_NAME: node["label"].strip(),
+            CONF_INPUT: node["props"]["Stream"],
+            CONF_EXTRA_ARGUMENTS: DEFAULT_ARGUMENTS,
+        }
+        FFmpegCamera.__init__(self, hass, device_info)
+
+        self._supported_features = (
+            CameraEntityFeature.ON_OFF | CameraEntityFeature.STREAM
+        )
+
+        self._command_motion_detection = self.get_command_id(
+            node["type"]["endpoints"], ATTR_DETECTION
+        )
+        self._command_flip = self.get_command_id(node["show_endpoints"], ATTR_FLIP)
+
+        self.update_node(node)
+
+    @property
+    def command_flip(self) -> Any | str:
+        """Return command_flip."""
+        return self._command_flip
+
+    def flip(self) -> bool:
+        """Return flip."""
+        return self._flip
+
+    async def async_flip(self, entity: FreeboxCamera) -> None:
+        """Flip the camera stream."""
+        self._flip = not entity.flip
+        await entity.set_home_endpoint_value(entity.command_flip, entity.flip)
+
+    async def async_enable_motion_detection(self) -> None:
+        """Enable motion detection in the camera."""
+        await self.set_home_endpoint_value(self._command_motion_detection, True)
+        self._attr_motion_detection_enabled = True
+
+    async def async_disable_motion_detection(self) -> None:
+        """Disable motion detection in camera."""
+        await self.set_home_endpoint_value(self._command_motion_detection, False)
+        self._attr_motion_detection_enabled = False
+
+    async def async_update_signal(self) -> None:
+        """Update the camera node."""
+        self.update_node(self._router.home_devices[self._id])
+
+    def update_node(self, node):
+        """Update params."""
+        self._name = node["label"].strip()
+
+        # Get status
+        if self._node["status"] == "active":
+            self._attr_is_streaming = True
+        else:
+            self._attr_is_streaming = False
+
+        # Parse all endpoints values & needed commands
+        self._attr_motion_detection_enabled = self.get_value("signal", ATTR_DETECTION)
+        self._activation_with_alarm = self.get_value("signal", ATTR_ACTIVATION)
+        self._high_quality_video = self.get_value("signal", ATTR_QUALITY)
+        self._motion_sensitivity = self.get_value("signal", ATTR_SENSITIVITY)
+        self._motion_threshold = self.get_value("signal", ATTR_THRESHOLD)
+        self._flip = self.get_value("signal", ATTR_FLIP)
+        self._timestamp = self.get_value("signal", ATTR_TIMESTAMP)
+        self._mic_volume = self.get_value("signal", ATTR_VOLUME)
+        self._sound_detection = self.get_value("signal", ATTR_SOUND_DETECTION)
+        self._sound_trigger = self.get_value("signal", ATTR_SOUND_TRIGGER)
+        self._rtsp = self.get_value("signal", ATTR_RTSP)
+        self._disk = self.get_value("signal", ATTR_DISK)
```

Comment:
Where are those values displayed/used ?

I think I see for `_attr_motion_detection_enabled`, but don't know for the rest.

Reply:
It's to get values from camera attributes ... because this attributes can be changed/modified from freebox application ... for example, it's possible set flip or timestamp values in freebox app and see the modifications in HA

These attributes are not static...so use extra_state_attributes is not a good idea

Reply:
I read the code, I understood why you are getting and updatting those attrs, it's not the question.

Here, you are fetching values, the question is: How the user is using those values ? How can he access it ?

The `extra_state_attributes` function car display mutable values.

Contact me on Discord if needed, that would speed up the process and code review : Quentame#7196

Reply:
Ok, so you chose to remove those unexposed attrs.